lib/auth/touchid/api.go

```diff
@@ -435,23 +435,11 @@ func Login(origin, user string, assertion *wanlib.CredentialAssertion) (*wanlib.
 	})
 
 	// Verify infos against allowed credentials, if any.
-	var cred *CredentialInfo
-	if len(assertion.Response.AllowedCredentials) > 0 {
-		for _, info := range infos {
-			for _, allowedCred := range assertion.Response.AllowedCredentials {
-				if info.CredentialID == string(allowedCred.CredentialID) {
-					cred = &info
```

Comment:
Was this line the capture-by-reference bug you mentioned in the description? The same that's [mentioned in common mistakes](https://github.com/golang/go/wiki/CommonMistakes#using-reference-to-loop-iterator-variable)?

I'm asking this because I'm still learning Go and want to be prepared for the inevitable day where I make this mistake. ;f

Reply:
Yep, that's the one. There are 2 subtle issues that compound into a bug - the `break` wasn't escaping the outer for and `info` is captured by reference, so it always ends up pointing to the last element in the `infos` array.

Like I said in that one Slack thread, this one bites everyone eventually. ;)